src/highdicom/sr/content.py

```diff
@@ -103,6 +105,34 @@ def __init__(
             relationship_type=RelationshipTypeValues.INFERRED_FROM
         )
 
+    @classmethod
+    def from_source_dataset(
```

Comment:
Given that we are considering adding a `from_dataset()`, it may be better to call this method just `from_source_image()`. What do you think?

P.S.: If we do this, we should apply the same logic to the other methods as well.

Reply:
Yeah, agreed. Changed for all relevant classes